app/models/miq_event.rb

```diff
@@ -154,4 +158,13 @@ def self.raise_event_for_children(target, raw_event, inputs = {})
   def self.event_name_for_target(target, event_suffix)
     "#{target.class.base_model.name.underscore}_#{event_suffix}"
   end
+
+  # return the event that triggered the policy event
+  def source_event
+    return @source_event if @source_event
+    return unless full_data
+
+    source_event_id = full_data.fetch_path(:source_event_id)
+    @source_event = EmsEvent.find_by(:id => source_event_id) if source_event_id
```

Comment:
Can any event_stream subclass be used to raise a evm_event?  If so the lookup should be against `EventStream` instead of `EmsEvent`.